conan/api/subapi/workspace.py

```diff
@@ -282,3 +282,53 @@ def collapse_editables(self, deps_graph, profile_host, profile_build):
             result.add_edge(root, t.node, r)
 
         return result
+
+    def export(self, lockfile=None, remotes=None):
+        self._check_ws()
+        exported = []
+        for ref, info in self.editable_packages.items():
+            exported_ref = self._conan_api.export.export(info["path"], None, None, None, None,
```

Comment:
```suggestion
            exported_ref = self._conan_api.export.export(info["path"], ref.name, str(ref.version), ref.user, ref.channel,
```

so that recipes that set the name/version in the package() method correctly work, otherwise you get `ERROR: conanfile didn't specify version`

Reply:
Which ``package()`` method? You mean the ``set_name/set_version()`` methods?
A recipe to be exported needs these fields, and the ``conan workspace create`` cannot pass all names/versions for all packages of the workspace, so necessarily, the ``conanfile`` of those packages need to provide their own name/version, it cannot be defined externally (as an argument)

Reply:
Done, both for ``workspace create + build``, with tests, please re-try